pkg/bugreport/ingress.go

```diff
@@ -0,0 +1,138 @@
+package bugreport
+
+import (
+	"context"
+	"path"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	nginxIngressControllerLabelSelector   string = "app.kubernetes.io/name=ingress-nginx"
+	contourIngressControllerLabelSelector string = "app.kubernetes.io/name=contour"
+
+	ingressResourceName        string = "ingresses"
+	ingressBackendResourceName string = "ingressbackends"
+)
+
+func (c *Config) collectIngressReport() error {
+	namespaceList, err := c.KubeClient.CoreV1().Namespaces().List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	for _, namespace := range namespaceList.Items {
+		ingressList, err := c.KubeClient.NetworkingV1().Ingresses(namespace.Name).List(context.Background(), metav1.ListOptions{})
+		if err != nil {
+			return err
+		}
+		for _, ingress := range ingressList.Items {
```

Comment:
I wonder if it would be prudent to (at some point in the future) carve this out as a new function called `getAllIngresses`, which could do just that - traverse all namespaces and look for Ingress resources. This would abstract away the 2 loops and error conditions.